examples/seq2seq/test_bash_script.py

```diff
@@ -3,92 +3,118 @@
 import argparse
 import os
 import sys
-from pathlib import Path
 from unittest.mock import patch
 
-import pytest
 import pytorch_lightning as pl
 import timeout_decorator
 import torch
 
 from distillation import BartSummarizationDistiller, distill_main
 from finetune import SummarizationModule, main
-from test_seq2seq_examples import CUDA_AVAILABLE, MBART_TINY
+from test_seq2seq_examples import MBART_TINY
 from transformers import BartForConditionalGeneration, MarianMTModel
-from transformers.testing_utils import TestCasePlus, slow
+from transformers.file_utils import cached_path
+from transformers.testing_utils import TestCasePlus, require_torch_gpu, slow
 from utils import load_json
 
 
 MODEL_NAME = MBART_TINY
-MARIAN_MODEL = "sshleifer/student_marian_en_ro_6_1"
+MARIAN_MODEL = "sshleifer/mar_enro_6_3_student"
 
 
 class TestAll(TestCasePlus):
+    def setUp(self):
+        super().setUp()
+        data_cached = cached_path(
+            "https://cdn-datasets.huggingface.co/translation/wmt_en_ro.tar.gz", extract_compressed_file=True
+        )
+        self.data_dir = f"{data_cached}/wmt_en_ro"
+
     @slow
-    @pytest.mark.skipif(not CUDA_AVAILABLE, reason="too slow to run on CPU")
+    @require_torch_gpu
     def test_model_download(self):
         """This warms up the cache so that we can time the next test without including download time, which varies between machines."""
         BartForConditionalGeneration.from_pretrained(MODEL_NAME)
         MarianMTModel.from_pretrained(MARIAN_MODEL)
 
-    @timeout_decorator.timeout(120)
+    # @timeout_decorator.timeout(1200)
     @slow
-    @pytest.mark.skipif(not CUDA_AVAILABLE, reason="too slow to run on CPU")
+    @require_torch_gpu
     def test_train_mbart_cc25_enro_script(self):
-        data_dir = "examples/seq2seq/test_data/wmt_en_ro"
+        DEBUG = 1  # XXX
+        data_dir = self.data_dir
         env_vars_to_replace = {
             "--fp16_opt_level=O1": "",
             "$MAX_LEN": 128,
```

Comment:
MAX_LEN=64 might speed things up?

Reply:
128: 3min, 64: 2:50m - the bleu scores for val/test are within 0.1 difference

Switch anyway as every second adds up?

Reply:
Yep!